crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -65,6 +65,22 @@ pub(super) enum EventTimelineItemKind {
     Remote(RemoteEventTimelineItem),
 }
 
+/// A wrapper that can contain either a transaction id, or an event id.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum EventItemIdentifier {
+    TransactionId(OwnedTransactionId),
+    EventId(OwnedEventId),
+}
+
+/// Data associated with a reaction sender.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ReactionSenderData {
+    /// Sender identifier.
+    pub id: OwnedUserId,
+    /// Date at which the sender reacted.
+    pub ts: MilliSecondsSinceUnixEpoch,
```

Comment:
```suggestion
    pub time: MilliSecondsSinceUnixEpoch,
```

or

```suggestion
    pub timestamp: MilliSecondsSinceUnixEpoch,
```

but please, don't use short names.